File: src/main/java/com/shion1305/ynufes/bodytemp2022/config/InstanceData.java
/*
 * Copyright (c) 2022 Shion Ichikawa All Rights Reserved.
 */

package com.shion1305.ynufes.bodytemp2022.config;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class InstanceData {
    public String processName;
    public String gasUrl;
    public String lineToken;
    public volatile boolean enabled;

    public InstanceData(@JsonProperty("name") String name, @JsonProperty("GASUrl") String getGasUrl, @JsonProperty("LineMessagingToken") String token, @JsonProperty("enabled") boolean enabled) {
        this.processName = name;
        this.gasUrl = getGasUrl;
        this.lineToken = token;
        this.enabled = enabled;
    }
}

File: src/main/java/com/shion1305/ynufes/bodytemp2022/config/InstanceData.kt
/*
 * Copyright (c) 2022 Shion Ichikawa All Rights Reserved.
 */
package com.shion1305.ynufes.bodytemp2022.config

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class InstanceData(
    @param:JsonProperty("name") var processName: String,
    @param:JsonProperty("GASUrl") var gasUrl: String,
    @param:JsonProperty(
        "LineMessagingToken"
    ) var lineToken: String,
    @field:Volatile @param:JsonProperty("enabled") var enabled: Boolean
)